argoproj-labs/argocd-operator
# Changing log level/format triggers reconciliation

**What type of PR is this?**
 /kind bug

**What does this PR do / why we need it**:
When changing the log level or log format, the operator should change the deployment spec even if the argocd-repo-server deployment resource already exists, currently that is not happening. 

**Have you updated the necessary documentation?**

N/A

**Which issue(s) this PR fixes**:

Fixes #448 

**How to test changes**:
1. Inspect `argocd-repo-server` deployment's container command line: `kubectl get deployment argocd-repo-server -o jsonpath='{.spec.template.spec.containers[0].command}'`. Observe `--loglevel info` and `--logformat text` command line parameters.
2. Patch the `ArgoCD` Operand to change log level and format for the repository server: `kubectl patch argocd argocd --type=merge --patch '{"spec": {"repo": {"logFormat": "json", "logLevel": "debug"}}}'`
3. Inspect `argocd-repo-server` deployment's container command line again: `kubectl get deployment argocd-repo-server -o jsonpath='{.spec.template.spec.containers[0].command}'`. Observe command line parameters have been changed from step 1. 

cc: @jannfis @jaideepr97
